### ISSUE TITLE
Linux/wayland: mouse position is off

### ISSUE THREAD

**Issue by johnae** [alacritty/alacritty#3205]
OS: Linux/BSD/macOS/Windows
Version: 0.4.2-dev (https://github.com/alacritty/alacritty/commit/12e211d82a1bdef4ffe70c74ee62bc56edfa7590)
Linux/BSD: Wayland, Sway

This seems like a similar issue to https://github.com/alacritty/alacritty/issues/3191 but that specifically says MacOS so I'm filing this for Linux/Wayland/Sway.

In my case I'm on a hidpi display and it seems related as I haven't noticed the same problem on an external, non-hidpi display.

**Comment by DivineGod** [comment]
I think this is something that's been fixed in [winit](https://github.com/rust-windowing/winit/commit/a6d180cefbc1b5046e324e033c93f529f416fdda) so we'll have to wait for an upstream winit release to test this out.

**Comment by DivineGod** [comment]
@johnae can you compile from source? If so you could test out adding

```toml
[patch.crates-io]
winit =  git = 'https://github.com/rust-windowing/winit' }
```

to the top-level Cargo.toml and re-build Alacritty

**Comment by chrisduerr** [comment]
I've renamed https://github.com/alacritty/alacritty/issues/3191, to include Wayland, so we can track both Wayland/macOS there.

The upstream fix has landed already, so I'll update #3191 as soon as a new `winit` version is released and things are fixed in Alacritty's master branch.
